sklearn/multioutput.py

```diff
@@ -64,13 +64,24 @@ def _partial_fit_estimator(
     return estimator
 
 
+def _available_if_base_estimator_has(attr):
+    def _check(self):
+        return (
+            (hasattr(self, "base_estimator") and hasattr(self.base_estimator, attr))
+            or hasattr(self.estimator, attr)
+            or all(hasattr(est, attr) for est in self.estimators_)
```

Comment:
This would make a difference between fitted and unfitted estimators, do we want that?

Reply:
It's what we're presently doing for metaestimators, because usually the properties are the same whether fitted or not, but there are edge cases where `predict_proba` capability can't be determined until after fitting.

Reply:
Interesting. Shouldn't it be that `predict_proba` is always there in those cases, but raises a relevant error if it can't produce the right output?